core/benches/scheduler.rs

```diff
@@ -0,0 +1,494 @@
+    }
+
+    fn run(
+        &self,
+        mut scheduler: impl Scheduler<Tx>,
+        mut container: TransactionStateContainer<Tx>,
+        stats: &mut BenchStats,
+    ) {
+        // each bench measurement is to schedule everything in the container
+        while !container.is_empty() {
+            // TODO - should call this, but it panics
+            // scheduler.receive_completed(&mut container).unwrap();
+
+            let result = scheduler
+                .schedule(&mut container, self.filter_1, self.filter_2)
+                .unwrap();
+
+            // do some VERY QUICK stats collecting to print/assert at end of bench
+            stats.num_of_scheduling = stats.num_of_scheduling.saturating_add(1);
+            stats.num_scheduled = stats.num_scheduled.saturating_add(result.num_scheduled);
+        }
+
+        stats.bench_iter_count = stats.bench_iter_count.saturating_add(1);
+    }
+}
+
+fn bench_prio_graph_scheuler(c: &mut Criterion) {
+    let mut stats = BenchStats::default();
+    let bench_env: BenchEnv<RuntimeTransaction<SanitizedTransaction>> = BenchEnv::new(&mut stats);
+
+    let mut group = c.benchmark_group("bench_prio_graph_scheduler_with_sdk_transactions");
+    group.sample_size(10);
+
+    let txs_builders: Vec<(&str, Box<dyn TestTxsBuilder>)> = vec![
+        ("simple_transfer", Box::new(SimpleTransferTxBuilder)),
+        ("many_transfer", Box::new(ManyTransfersTxBuilder)),
+        ("packed_noop", Box::new(PackedNoopsTxBuilder)),
+    ];
+
+    let conflict_conditions: Vec<(&str, bool)> =
+        vec![("non-conflict", false), ("full-conflict", true)];
+
+    let tx_counts: Vec<(&str, usize)> = vec![
+        ("empty_container", 0),
+        ("full_container", TOTAL_BUFFERED_PACKETS),
+    ];
+
+    for (txs_builder_type, txs_builder) in &txs_builders {
+        for (conflict_condition_type, conflict_condition) in &conflict_conditions {
+            for (tx_count_type, tx_count) in &tx_counts {
+                let bench_name =
+                    format!("{txs_builder_type}/{conflict_condition_type}/{tx_count_type}");
+                group.throughput(Throughput::Elements(*tx_count as u64));
+                group.bench_function(&bench_name, |bencher| {
+                    bencher.iter_custom(|iters| {
+                        let mut execute_time: Duration = std::time::Duration::ZERO;
+                        for _i in 0..iters {
+                            // setup new Scheduler and Container for each iteration of execution
+                            let (scheduler, container) = {
+                                let mut bench_container =
+                                    BenchContainer::new(TOTAL_BUFFERED_PACKETS);
+                                bench_container.fill_container(
+                                    txs_builder
+                                        .build(*tx_count, *conflict_condition)
+                                        .into_iter(),
```

Comment:
nit - we call `build` and then immediately `into_iter`. `build` could just return an iterator since we don't need to actually `collect` into a vector.

Reply:
At first I tried `build() ->impl iter<> { std::iter.once(tracer).chain((1..count).map(...)) }`, but when dynamically casting different tx builders at ln364, compiler complaint about trait `TestTxsBuilder` isn't dyn-compatible due to returning `impl iter`. I didn't look further into other solutions, but to return vec.

Reply:
this turns `TestTxBuilder` into "instructions builder"; Would like to have a `TxBuilder` that has polymorphism that builds both sdk and view type txs (that supports TransactionWithMeta trait?). This subject directly leads to next step - builder supports transaction_view. We can either support that in tis pr, or do that in next PR. wdyt @apfitzge @KirillLykov

Reply:
@KirillLykov and I planned to reuse transaction generation functions between receive_and_buffer bench and scheduler bench. I'm giving it a stab at fe5332d716742533af4c5c94a194d47279987e5d and d70d29f2b683ed653acef5a54c76f76b8bbacec5 to streamline this part of code. I like how it turns out to be, wdyt?